proxy/src/main/java/com/wavefront/agent/PlainTextOrHttpFrameDecoder.java

```diff
@@ -66,13 +74,23 @@ protected void decode(final ChannelHandlerContext ctx,
 
     // determine the protocol and add the encoder/decoder
     final ChannelPipeline pipeline = ctx.pipeline();
-    if (isHttp(firstByte, secondByte)) {
-      pipeline.addLast("decoder", new HttpRequestDecoder());
-      pipeline.addLast("encoder", new HttpResponseEncoder());
-      pipeline.addLast("aggregator", new HttpObjectAggregator(16 * 1024 * 1024));
-      pipeline.addLast("deflate", new HttpContentCompressor());
-      pipeline.addLast("handler", this.handler);
+
+    if (detectGzip && isGzip(firstByte, secondByte)) {
+      logger.fine("Inbound gzip stream detected");
+      pipeline
+          .addLast("gzipdeflater", ZlibCodecFactory.newZlibEncoder(ZlibWrapper.GZIP))
```

Comment:
should any of the encoders gzip and http be cached?

Reply:
In Netty everything http and gzip-related is not stateless, unfortunately, so these objects can't be shared.